packages/runtime/test/cli/watch.test.mjs

```diff
@@ -124,11 +124,84 @@ test('should not hot reload files with `--hot-reload false', async (t) => {
   t.after(() => child.kill('SIGINT'))
-  child.stdout.pipe(process.stderr)
-  child.stderr.pipe(process.stderr)
   await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v2', true))
   await sleep(5000)
   const res = await request(`${url}/version`)
   const version = await res.body.text()
   assert.strictEqual(version, 'v1')
 })
+
+test('watches CommonJS files with hotreload', { timeout: 30000, skip: process.env.CI }, async (t) => {
+  const tmpDir = await mkdtemp(join(base, 'watch-'))
+  t.after(() => saferm(tmpDir))
+  t.diagnostic(`using ${tmpDir}`)
+  const configFileSrc = join(fixturesDir, 'configs', 'hotreload.json')
+  const configFileDst = join(tmpDir, 'configs', 'monorepo.json')
+  const appSrc = join(fixturesDir, 'monorepo')
+  const appDst = join(tmpDir, 'monorepo')
+  const cjsPluginFilePath = join(appDst, 'serviceAppWithLogger', 'plugin.js')
+
+  await Promise.all([
+    cp(configFileSrc, configFileDst),
+    cp(appSrc, appDst, { recursive: true })
+  ])
+
+  await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v1', false))
+  const { child } = await start('-c', configFileDst)
+  t.after(() => child.kill('SIGINT'))
+
+  await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v2', true))
+
+  let restartedSecondTime = false
+  let restartedThirdTime = false
+
+  for await (const log of child.ndj) {
+    if (log.msg === 'RELOADED v2') {
+      restartedSecondTime = true
+    } else if (log.msg === 'RELOADED v3') {
+      restartedThirdTime = true
+      break
+    } else if (log.msg?.match(/watching/)) {
+      await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v3', true))
+    }
+  }
+
+  assert.ok(restartedSecondTime)
+  assert.ok(restartedThirdTime)
+})
+
+test('watches CommonJS files with hotreload on a single service', { timeout: 30000, skip: process.env.CI }, async (t) => {
+  const tmpDir = await mkdtemp(join(base, 'watch-'))
+  t.after(() => saferm(tmpDir))
+  t.diagnostic(`using ${tmpDir}`)
+  const appSrc = join(fixturesDir, 'monorepo', 'serviceAppWithLogger')
+  const appDst = join(tmpDir)
+  const cjsPluginFilePath = join(appDst, 'plugin.js')
+
+  await Promise.all([
+    cp(appSrc, appDst, { recursive: true })
+  ])
+
+  await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v1', false))
+  const { child } = await start('-c', join(appDst, 'platformatic.service.json'))
+  t.after(() => child.kill('SIGINT'))
+
+  await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v2', true))
+
+  let restartedSecondTime = false
+  let restartedThirdTime = false
+
+  for await (const log of child.ndj) {
+    if (log.msg === 'RELOADED v2') {
+      restartedSecondTime = true
+    } else if (log.msg === 'RELOADED v3') {
+      restartedThirdTime = true
+      break
+    } else if (log.msg?.match(/listening/)) {
+      await writeFile(cjsPluginFilePath, createCjsLoggingPlugin('v3', true))
+    }
+  }
+
+  assert.ok(restartedSecondTime)
```

Comment:
I would move this to line 198 because then we are also ensuring that the ordering of the events is correct.